### DISCUSSION TITLE
Type 'TreeNode<any> | TreeNode<any>[] | null' is not assignable to type 'TreeNode<any>'.   Type 'null' is not assignable to type 'TreeNode<any>

### DISCUSSION THREAD

**Discussion by vrjaroslav** [primefaces/community#512] [PrimeNG]
`<p-tree selectionMode="single" [(selection)]="selectedTreeNode">`

in ts:  selectedTreeNode: TreeNode;

I am getting error: 
**Error: error TS2322: Type 'TreeNode<any> | TreeNode<any>[] | null' is not assignable to type 'TreeNode<any>'.
  Type 'null' is not assignable to type 'TreeNode<any>'.**

I use primeng version 16.8.0 and angular version: 17.0.2

I found something here, [https://github.com/primefaces/primeng/issues/13183]( https://github.com/primefaces/primeng/issues/13183 ), but it does not help me

Can anybody please help me ?

**Comment by LangOliver** [comment]
Had the same issue, not sure where the underlying problem is, but I would be interested to know.
Nevertheless, a quick fix would be `selectedTreeNode!: any;`
Kind regards, Oli

**Comment by vrjaroslav** [comment]
Thanks @LangOliver for reply and suggested fix. That fix woks for me fine. 
It means that I have to change the code and if I want work with selectedTreeNode.

I had in code: `selectedTreeNode.value`, but after I changed the type to any, then I need to change it as for use Intellisense:

`(selectedTreeNode as TreeNode).value` 

I hope someone from PrimeNg team will fix this bug. I think it is there since July this year.

**Comment by cetincakiroglu** [comment]
Hi,

We've made a type enhancenments in v16, this issue probably happened because of that iteration. Could you please share a stackblitz example so we can address this issue in the upcoming releases?

**Reply by vrjaroslav** [reply]
Hi @cetincakiroglu, thank you for your answer. I just created new Stackblitz, it is based on sample from PrimeNg documentation.
I only added to file tsconfig.json this section:

`"angularCompilerOptions": {
    "strictTemplates": true
  }`

without it it works good, but with this section it throw exception. 
You can find Stackblitz here: [https://stackblitz.com/edit/p5t4vm-oe54gn?file=tsconfig.json](https://stackblitz.com/edit/p5t4vm-oe54gn?file=tsconfig.json)
